src/compile/mark/valueref.ts

```diff
@@ -216,13 +217,15 @@ export function tooltipForChannelDefs(channelDefs: FieldDef<string>[], config: C
   return keyValues.length ? {signal: `{${keyValues.join(', ')}}`} : undefined;
 }
 
-export function text(textDef: ChannelDefWithCondition<TextFieldDef<string>>, config: Config): VgValueRef {
+export function text(channelDef: ChannelDefWithCondition<TextFieldDef<string>>, config: Config): VgValueRef {
   // text
-  if (textDef) {
-    if (isFieldDef(textDef)) {
-      return formatSignalRef(textDef, textDef.format, 'datum', config);
-    } else if (isValueDef(textDef)) {
-      return {value: textDef.value};
+  if (channelDef) {
+    if (isValueDef(channelDef)) {
+      return {value: channelDef.value};
+    }
+    const formatString = isFieldDef(channelDef) ? format(channelDef) : undefined;
```

Comment:
Why checking isFieldDef multiple time & swapping the condition order? 

I think the code should be pretty much the same as the original code except replacing `textDef.format` with ` format(channelDef)`.

There is also no need to rename the variable `textDef`.

Reply:
Actually renaming textDef to channelDef makes sense, but you have to change the type to be general FieldDef too.